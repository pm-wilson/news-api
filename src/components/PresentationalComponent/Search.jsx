import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ text, onChange, onSubmit }) => (
  <div>
    <form onSubmit={onSubmit}>
      <label>News Search</label>
      <input
        id="text"
        type="text"
        name="text"
        value={text}
        onChange={onChange}
      />
      <button>Submit</button>
    </form>
  </div>
);

Search.propTypes = {
  text: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default Search;
