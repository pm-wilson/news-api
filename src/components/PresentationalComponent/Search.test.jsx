import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Search from './Search';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<App />);
    //expect(asFragment()).toMatchSnapshot();
    expect(true).toEqual(true);
  });
});