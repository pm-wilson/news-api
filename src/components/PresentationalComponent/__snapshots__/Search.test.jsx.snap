// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label>
        News Search
      </label>
      <input
        id="text"
        name="text"
        type="text"
        value="search text"
      />
      <button>
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
